File: frontend/src/entities/crm-communication/dto/crm-communication.dto.ts
import { CommunicationType, CommunicationDirection } from "@entities/crm-communication/models/crm-communication.model";

export interface CrmCommunicationDto {
  contactId?: number;
  clientId?: number;
  type: CommunicationType;
  direction?: CommunicationDirection;
  subject?: string;
  content?: string;
}

File: src/main/java/io/hearstcorporation/atelier/dto/model/crm/CrmCommunicationDto.java
package io.hearstcorporation.atelier.dto.model.crm;

import io.hearstcorporation.atelier.model.crm.CommunicationDirection;
import io.hearstcorporation.atelier.model.crm.CommunicationType;
import lombok.Builder;
import lombok.Getter;

import java.time.Instant;

@Getter
@Builder
public class CrmCommunicationDto {
    private Long id;
    private Long contactId;
    private String contactName;
    private Long clientId;
    private String clientName;
    private CommunicationType type;
    private CommunicationDirection direction;
    private String subject;
    private String content;
    private Instant sentAt;
    private Long createdById;
    private String createdByName;
    private Instant createdAt;
}

